EIPS/eip-7783.md

```diff
@@ -0,0 +1,99 @@
+---
+eip: 7783
+title: Add Controlled Gas Target Increase Strategy
+description: Adds a controlled gas target increase strategy.
+author: Giulio Rebuffo (@Giulio2002)
+discussions-to: https://ethereum-magicians.org/t/eip-7783-add-controlled-gas-target-increase-strategy/21282
+status: Draft
+type: Informational
+created: 2024-10-06
+---
+
+This proposal describes the introduction in clients of a controlled gas target increase strategy to determine the gas limit of a specific block and set as default with conservative parameters, while keeping the possibility to change it in the future to a fixed value.
+
+## **Abstract**
+
+The EIP proposes the introduction of a new gas target management mechanism that automatically increases the block gas target over time. The incremental growth is controlled by a fixed rate, ensuring predictable network scaling while preventing sudden surges in block sizes. This strategy is meant to be used as a default setting, with the option to switch to a fixed gas target if needed (or different parameters).
+
+## **Motivation**
+
+### **Predictable Gas Target Growth**
+
+- **Current Issue:**
+  - The Ethereum network faces increasing demand, but changes to the gas target are often manually adjusted by miners or validators based on their preferences, which may cause unpredictable block sizes and network performance issues.
+
+- **Need for Change:**
+  - A systematic and predictable increase of the gas target will help scale the network while giving the ecosystem time to adjust to larger block sizes, without needing to rely on ad hoc decisions by network participants.
+
+### **Gradual Increase with Deactivation Safeguard**
+
+- **Controlled Growth:**
+  - Instead of sudden or unpredictable changes, this EIP proposes incremental gas Target increases over a specified amount of time, ensuring a smooth transition to higher transaction throughput, while still keeping the governance of the gas Target in the hand of the community.
+  
+- **Automatic Deactivation:**
+  - A safeguard deactivation block will halt the increase after some specified time, preventing the gas Target from growing indefinitely and allowing the community to reassess the network's needs.
+
+## **Specification**
+
+### **Incremental Gas Target Increase Strategy**
+
+Add a new "Gas Target" selection strategy that takes in Block Number `N` and spits out the Gas Target `GT` for that block. The strategy is as follows:
+
+ - The gas target `GT_t` at block `t` is calculated as:
+
+```python
+def compute_gas_target(blockNum: int, blockNumStart: int, initialGasTarget: int, r: int, gasTargetCap: int) -> int:
```

Comment:
and gasLimit stays same or limit also grows with the 2X relation?

Reply:
if gasLimit is also changing in same proportion, then `2r` has to be less < gasLimit/1024 which is easily statisfied i guess, but still needs a mention for context of `r`

Reply:
yes - if the gas target is increased in a large amount sample of validators, yes

Reply:
I mentioned that gas limit still grows 2x in proportion to the gas target. tell me if it is satisfiable